testsuite/pytests/test_synapsecollection.py

```diff
@@ -363,15 +363,16 @@ def test_empty(self):
         """
         conns = nest.GetConnections()
         self.assertEqual(len(conns), 0)
-        self.assertEqual(conns.get(), ())
+        self.assertEqual(conns.get(), {})
+        self.assertEqual(conns.get('weight'), ())
 
         nrns = nest.Create('iaf_psc_alpha', 2)
         nest.Connect(nrns, nrns, 'one_to_one')
         conns = nest.GetConnections()
         self.assertEqual(len(conns), 2)
 
         nest.ResetKernel()
-        self.assertEqual(conns.get(), ())
+        self.assertEqual(conns.get(), {})
         conns.set(weight=10.)
```

Comment:
Is not resetting the `kernel` equivalent to adding or deleting `connections`, and in that case we should get a warning from the `ConnectionManager` stating a similar `info` as the following: `ConnectionManager [Warning]: 
    New connections created, connection descriptors previously obtained using 
    'GetConnections' are now invalid`. In that case, the user should know that setting the `weight` in `conn` object won't be a valid operation, or maybe throw an error that  the `weight` isn't in the connection dictionary anymore.